src/ini.pest

```diff
@@ -1,17 +1,24 @@
-ini = { SOI ~ bom? ~ (pair | comment | NEWLINE)* ~ section* ~ EOI }
-bom = _{ "\u{feff}" }
-WHITESPACE = _{ "\t" }
+ini = { SOI ~ (blank | pair | comment)* ~ section* }
 
-// section
-section = { " "* ~ "[" ~ name ~ "]" ~ " "* ~ body }
-name = @{ char+ }
-char = { ASCII_ALPHANUMERIC | "." | "_" | "/" | " " | ";" | "#" | "\\" }
-body = { (NEWLINE ~ (pair | comment)?)* }
+// lines
+blank = _{ NEWLINE }
+pair = { key ~ "=" ~ value ~ eol}
+comment = { indicator ~ value? ~ eol}
 
-// key-value pair
-pair = { key ~ "=" ~ value }
-key = @{ char{1,50} }
-value = @{ (char | "="){0,255} }
+// sections
+section = { "[" ~ name ~ "]" ~ eol ~ body? }
+body = { (blank | pair | comment)* }
 
-comment = { indicator ~ " "* ~ value? }
+// tokens
+// a name must end with a closing bracket followed by EOL
+name = @{ (!("]" ~ eol) ~ char)+ }
```

Comment:
this doesn't match `a` in `[a]  `. For some reason, the trailing whitespace _outside_ of the section is affecting it.